### ISSUE TITLE
spark-sql: Cannot connect to Hive Metastore

### ISSUE THREAD

**Issue by Nuttymoon** [TOSIT-IO/tdp-collection#402]
While `pyspark(3)` and `spark(3)-shell` can successfully interact with the Hive Metastore after #401, the `spark(3)-sql` CLI cannot connect to the Hive Metastore. We have the following error:
```
2022-08-03 09:37:51,153 WARN hive.metastore: Failed to connect to the MetaStore Server...
...
Caused by: java.lang.RuntimeException: Unable to instantiate org.apache.hadoop.hive.ql.metadata.SessionHiveMetaStoreClient
	at org.apache.hadoop.hive.metastore.MetaStoreUtils.newInstance(MetaStoreUtils.java:1523)
	at org.apache.hadoop.hive.metastore.RetryingMetaStoreClient.<init>(RetryingMetaStoreClient.java:86)
	at org.apache.hadoop.hive.metastore.RetryingMetaStoreClient.getProxy(RetryingMetaStoreClient.java:132)
	at org.apache.hadoop.hive.metastore.RetryingMetaStoreClient.getProxy(RetryingMetaStoreClient.java:104)
	at org.apache.hadoop.hive.ql.metadata.Hive.createMetaStoreClient(Hive.java:3005)
	at org.apache.hadoop.hive.ql.metadata.Hive.getMSC(Hive.java:3024)
	at org.apache.hadoop.hive.ql.session.SessionState.start(SessionState.java:503)
	... 12 more
...
Caused by: MetaException(message:Could not connect to meta store using any of the URIs provided. Most recent failure: org.apache.thrift.transport.TTransportException: Invalid status 21
```

The error occurs at different steps between Spark 2 and Spark 3:
- For `spark-sql`, it fails at instantiation
- For `spark3-sql`, it fails when trying to read a table:
  ```
  Hive Session ID = 6de6d09a-2733-4f52-a5bb-8bae91afc52b
  Spark master: yarn, Application Id: application_1659513794139_0021
  spark-sql> show tables in tdp_user_db;
  table1
  Time taken: 1.715 seconds, Fetched 1 row(s)
  spark-sql> select * from tdp_user_db.table1;
  2022-08-03 09:42:13,301 WARN session.SessionState: METASTORE_FILTER_HOOK will be ignored, since hive.security.authorization.manager is set to instance of HiveAuthorizerFactory.
  2022-08-03 09:42:13,374 WARN hive.metastore: Failed to connect to the MetaStore Server...
  2022-08-03 09:42:18,381 WARN hive.metastore: Failed to connect to the MetaStore Server...
  2022-08-03 09:42:23,388 WARN hive.metastore: Failed to connect to the MetaStore Server...
  2022-08-03 09:42:28,434 ERROR session.SessionState: Error setting up authorization: Unable to instantiate org.apache.hadoop.hive.ql.metadata.SessionHiveMetaStoreClient
  ```

There are probably some missing properties in the Spark `hive-site.xml`. We can see those warnings when running `spark-sql`:
```
2022-08-03 09:37:50,942 WARN conf.HiveConf: HiveConf of name hive.metastore.uri.selection does not exist
2022-08-03 09:37:50,949 WARN conf.HiveConf: HiveConf of name hive.metastore.truststore.path does not exist
2022-08-03 09:37:50,950 WARN conf.HiveConf: HiveConf of name hive.metastore.truststore.password does not exist
2022-08-03 09:37:50,950 WARN conf.HiveConf: HiveConf of name hive.metastore.use.SSL does not exist
```

**Comment by mehdibn** [comment]
@Nuttymoon thank you to open this issue that I did never meet before. 
I confirm having the same error in my env.
Do you want me to investigate more, or you are doing it ?

**Comment by mehdibn** [comment]
it is related to the **ssl conf** of `spark-sql` command as you can see in the `metastore.log` :
```
2022-08-03T14:48:52,748 ERROR [pool-8-thread-31] server.TThreadPoolServer: Error occurred during processing of message.
java.lang.RuntimeException: org.apache.thrift.transport.TTransportException: javax.net.ssl.SSLException: Unsupported or unrecognized SSL message
        at org.apache.thrift.transport.TSaslServerTransport$Factory.getTransport(TSaslServerTransport.java:219) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.hadoop.hive.metastore.security.HadoopThriftAuthBridge$Server$TUGIAssumingTransportFactory$1.run(HadoopThriftAuthBridge.java:694) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.hadoop.hive.metastore.security.HadoopThriftAuthBridge$Server$TUGIAssumingTransportFactory$1.run(HadoopThriftAuthBridge.java:691) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_332]
        at javax.security.auth.Subject.doAs(Subject.java:360) ~[?:1.8.0_332]
        at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1709) ~[hadoop-common-3.1.1-TDP-0.1.0-SNAPSHOT.jar:?]
        at org.apache.hadoop.hive.metastore.security.HadoopThriftAuthBridge$Server$TUGIAssumingTransportFactory.getTransport(HadoopThriftAuthBridge.java:691) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.server.TThreadPoolServer$WorkerProcess.run(TThreadPoolServer.java:269) [hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_332]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_332]
        at java.lang.Thread.run(Thread.java:750) [?:1.8.0_332]
Caused by: org.apache.thrift.transport.TTransportException: javax.net.ssl.SSLException: Unsupported or unrecognized SSL message
        at org.apache.thrift.transport.TIOStreamTransport.read(TIOStreamTransport.java:129) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TTransport.readAll(TTransport.java:86) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslTransport.receiveSaslMessage(TSaslTransport.java:178) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslServerTransport.handleSaslStartMessage(TSaslServerTransport.java:125) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslTransport.open(TSaslTransport.java:271) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslServerTransport.open(TSaslServerTransport.java:41) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslServerTransport$Factory.getTransport(TSaslServerTransport.java:216) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        ... 10 more
Caused by: javax.net.ssl.SSLException: Unsupported or unrecognized SSL message
        at sun.security.ssl.SSLSocketInputRecord.handleUnknownRecord(SSLSocketInputRecord.java:455) ~[?:1.8.0_332]
        at sun.security.ssl.SSLSocketInputRecord.decode(SSLSocketInputRecord.java:184) ~[?:1.8.0_332]
        at sun.security.ssl.SSLTransport.decode(SSLTransport.java:109) ~[?:1.8.0_332]
        at sun.security.ssl.SSLSocketImpl.decode(SSLSocketImpl.java:1397) ~[?:1.8.0_332]
        at sun.security.ssl.SSLSocketImpl.readHandshakeRecord(SSLSocketImpl.java:1305) ~[?:1.8.0_332]
        at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:440) ~[?:1.8.0_332]
        at sun.security.ssl.SSLSocketImpl.ensureNegotiated(SSLSocketImpl.java:818) ~[?:1.8.0_332]
        at sun.security.ssl.SSLSocketImpl.access$200(SSLSocketImpl.java:73) ~[?:1.8.0_332]
        at sun.security.ssl.SSLSocketImpl$AppInputStream.read(SSLSocketImpl.java:909) ~[?:1.8.0_332]
        at java.io.BufferedInputStream.fill(BufferedInputStream.java:246) ~[?:1.8.0_332]
        at java.io.BufferedInputStream.read1(BufferedInputStream.java:286) ~[?:1.8.0_332]
        at java.io.BufferedInputStream.read(BufferedInputStream.java:345) ~[?:1.8.0_332]
        at org.apache.thrift.transport.TIOStreamTransport.read(TIOStreamTransport.java:127) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TTransport.readAll(TTransport.java:86) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslTransport.receiveSaslMessage(TSaslTransport.java:178) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslServerTransport.handleSaslStartMessage(TSaslServerTransport.java:125) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslTransport.open(TSaslTransport.java:271) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslServerTransport.open(TSaslServerTransport.java:41) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslServerTransport$Factory.getTransport(TSaslServerTransport.java:216) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        ... 10 more
2022-08-03T14:48:53,766 ERROR [pool-8-thread-32] server.TThreadPoolServer: Error occurred during processing of message.
java.lang.RuntimeException: org.apache.thrift.transport.TTransportException: javax.net.ssl.SSLException: Unsupported or unrecognized SSL message
        at org.apache.thrift.transport.TSaslServerTransport$Factory.getTransport(TSaslServerTransport.java:219) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.hadoop.hive.metastore.security.HadoopThriftAuthBridge$Server$TUGIAssumingTransportFactory$1.run(HadoopThriftAuthBridge.java:694) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.hadoop.hive.metastore.security.HadoopThriftAuthBridge$Server$TUGIAssumingTransportFactory$1.run(HadoopThriftAuthBridge.java:691) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_332]
        at javax.security.auth.Subject.doAs(Subject.java:360) ~[?:1.8.0_332]
        at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1709) ~[hadoop-common-3.1.1-TDP-0.1.0-SNAPSHOT.jar:?]
        at org.apache.hadoop.hive.metastore.security.HadoopThriftAuthBridge$Server$TUGIAssumingTransportFactory.getTransport(HadoopThriftAuthBridge.java:691) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.server.TThreadPoolServer$WorkerProcess.run(TThreadPoolServer.java:269) [hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_332]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_332]
        at java.lang.Thread.run(Thread.java:750) [?:1.8.0_332]
Caused by: org.apache.thrift.transport.TTransportException: javax.net.ssl.SSLException: Unsupported or unrecognized SSL message
        at org.apache.thrift.transport.TIOStreamTransport.read(TIOStreamTransport.java:129) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TTransport.readAll(TTransport.java:86) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslTransport.receiveSaslMessage(TSaslTransport.java:178) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslServerTransport.handleSaslStartMessage(TSaslServerTransport.java:125) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslTransport.open(TSaslTransport.java:271) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslServerTransport.open(TSaslServerTransport.java:41) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslServerTransport$Factory.getTransport(TSaslServerTransport.java:216) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        ... 10 more
Caused by: javax.net.ssl.SSLException: Unsupported or unrecognized SSL message
        at sun.security.ssl.SSLSocketInputRecord.handleUnknownRecord(SSLSocketInputRecord.java:455) ~[?:1.8.0_332]
        at sun.security.ssl.SSLSocketInputRecord.decode(SSLSocketInputRecord.java:184) ~[?:1.8.0_332]
        at sun.security.ssl.SSLTransport.decode(SSLTransport.java:109) ~[?:1.8.0_332]
        at sun.security.ssl.SSLSocketImpl.decode(SSLSocketImpl.java:1397) ~[?:1.8.0_332]
        at sun.security.ssl.SSLSocketImpl.readHandshakeRecord(SSLSocketImpl.java:1305) ~[?:1.8.0_332]
        at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:440) ~[?:1.8.0_332]
        at sun.security.ssl.SSLSocketImpl.ensureNegotiated(SSLSocketImpl.java:818) ~[?:1.8.0_332]
        at sun.security.ssl.SSLSocketImpl.access$200(SSLSocketImpl.java:73) ~[?:1.8.0_332]
        at sun.security.ssl.SSLSocketImpl$AppInputStream.read(SSLSocketImpl.java:909) ~[?:1.8.0_332]
        at java.io.BufferedInputStream.fill(BufferedInputStream.java:246) ~[?:1.8.0_332]
        at java.io.BufferedInputStream.read1(BufferedInputStream.java:286) ~[?:1.8.0_332]
        at java.io.BufferedInputStream.read(BufferedInputStream.java:345) ~[?:1.8.0_332]
        at org.apache.thrift.transport.TIOStreamTransport.read(TIOStreamTransport.java:127) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TTransport.readAll(TTransport.java:86) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslTransport.receiveSaslMessage(TSaslTransport.java:178) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslServerTransport.handleSaslStartMessage(TSaslServerTransport.java:125) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslTransport.open(TSaslTransport.java:271) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslServerTransport.open(TSaslServerTransport.java:41) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslServerTransport$Factory.getTransport(TSaslServerTransport.java:216) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        ... 10 more
2022-08-03T14:48:54,783 ERROR [pool-8-thread-33] server.TThreadPoolServer: Error occurred during processing of message.
java.lang.RuntimeException: org.apache.thrift.transport.TTransportException: javax.net.ssl.SSLException: Unsupported or unrecognized SSL message
        at org.apache.thrift.transport.TSaslServerTransport$Factory.getTransport(TSaslServerTransport.java:219) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.hadoop.hive.metastore.security.HadoopThriftAuthBridge$Server$TUGIAssumingTransportFactory$1.run(HadoopThriftAuthBridge.java:694) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.hadoop.hive.metastore.security.HadoopThriftAuthBridge$Server$TUGIAssumingTransportFactory$1.run(HadoopThriftAuthBridge.java:691) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_332]
        at javax.security.auth.Subject.doAs(Subject.java:360) ~[?:1.8.0_332]
        at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1709) ~[hadoop-common-3.1.1-TDP-0.1.0-SNAPSHOT.jar:?]
        at org.apache.hadoop.hive.metastore.security.HadoopThriftAuthBridge$Server$TUGIAssumingTransportFactory.getTransport(HadoopThriftAuthBridge.java:691) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.server.TThreadPoolServer$WorkerProcess.run(TThreadPoolServer.java:269) [hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_332]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_332]
        at java.lang.Thread.run(Thread.java:750) [?:1.8.0_332]
Caused by: org.apache.thrift.transport.TTransportException: javax.net.ssl.SSLException: Unsupported or unrecognized SSL message
        at org.apache.thrift.transport.TIOStreamTransport.read(TIOStreamTransport.java:129) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TTransport.readAll(TTransport.java:86) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslTransport.receiveSaslMessage(TSaslTransport.java:178) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslServerTransport.handleSaslStartMessage(TSaslServerTransport.java:125) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslTransport.open(TSaslTransport.java:271) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslServerTransport.open(TSaslServerTransport.java:41) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslServerTransport$Factory.getTransport(TSaslServerTransport.java:216) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        ... 10 more
Caused by: javax.net.ssl.SSLException: Unsupported or unrecognized SSL message
        at sun.security.ssl.SSLSocketInputRecord.handleUnknownRecord(SSLSocketInputRecord.java:455) ~[?:1.8.0_332]
        at sun.security.ssl.SSLSocketInputRecord.decode(SSLSocketInputRecord.java:184) ~[?:1.8.0_332]
        at sun.security.ssl.SSLTransport.decode(SSLTransport.java:109) ~[?:1.8.0_332]
        at sun.security.ssl.SSLSocketImpl.decode(SSLSocketImpl.java:1397) ~[?:1.8.0_332]
        at sun.security.ssl.SSLSocketImpl.readHandshakeRecord(SSLSocketImpl.java:1305) ~[?:1.8.0_332]
        at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:440) ~[?:1.8.0_332]
        at sun.security.ssl.SSLSocketImpl.ensureNegotiated(SSLSocketImpl.java:818) ~[?:1.8.0_332]
        at sun.security.ssl.SSLSocketImpl.access$200(SSLSocketImpl.java:73) ~[?:1.8.0_332]
        at sun.security.ssl.SSLSocketImpl$AppInputStream.read(SSLSocketImpl.java:909) ~[?:1.8.0_332]
        at java.io.BufferedInputStream.fill(BufferedInputStream.java:246) ~[?:1.8.0_332]
        at java.io.BufferedInputStream.read1(BufferedInputStream.java:286) ~[?:1.8.0_332]
        at java.io.BufferedInputStream.read(BufferedInputStream.java:345) ~[?:1.8.0_332]
        at org.apache.thrift.transport.TIOStreamTransport.read(TIOStreamTransport.java:127) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TTransport.readAll(TTransport.java:86) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslTransport.receiveSaslMessage(TSaslTransport.java:178) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslServerTransport.handleSaslStartMessage(TSaslServerTransport.java:125) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslTransport.open(TSaslTransport.java:271) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslServerTransport.open(TSaslServerTransport.java:41) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslServerTransport$Factory.getTransport(TSaslServerTransport.java:216) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        ... 10 more
2022-08-03T14:48:55,820 ERROR [pool-8-thread-34] server.TThreadPoolServer: Error occurred during processing of message.
java.lang.RuntimeException: org.apache.thrift.transport.TTransportException: javax.net.ssl.SSLException: Unsupported or unrecognized SSL message
        at org.apache.thrift.transport.TSaslServerTransport$Factory.getTransport(TSaslServerTransport.java:219) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.hadoop.hive.metastore.security.HadoopThriftAuthBridge$Server$TUGIAssumingTransportFactory$1.run(HadoopThriftAuthBridge.java:694) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.hadoop.hive.metastore.security.HadoopThriftAuthBridge$Server$TUGIAssumingTransportFactory$1.run(HadoopThriftAuthBridge.java:691) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_332]
        at javax.security.auth.Subject.doAs(Subject.java:360) ~[?:1.8.0_332]
        at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1709) ~[hadoop-common-3.1.1-TDP-0.1.0-SNAPSHOT.jar:?]
        at org.apache.hadoop.hive.metastore.security.HadoopThriftAuthBridge$Server$TUGIAssumingTransportFactory.getTransport(HadoopThriftAuthBridge.java:691) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.server.TThreadPoolServer$WorkerProcess.run(TThreadPoolServer.java:269) [hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_332]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_332]
        at java.lang.Thread.run(Thread.java:750) [?:1.8.0_332]
Caused by: org.apache.thrift.transport.TTransportException: javax.net.ssl.SSLException: Unsupported or unrecognized SSL message
        at org.apache.thrift.transport.TIOStreamTransport.read(TIOStreamTransport.java:129) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TTransport.readAll(TTransport.java:86) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslTransport.receiveSaslMessage(TSaslTransport.java:178) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslServerTransport.handleSaslStartMessage(TSaslServerTransport.java:125) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslTransport.open(TSaslTransport.java:271) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslServerTransport.open(TSaslServerTransport.java:41) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslServerTransport$Factory.getTransport(TSaslServerTransport.java:216) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        ... 10 more
Caused by: javax.net.ssl.SSLException: Unsupported or unrecognized SSL message
        at sun.security.ssl.SSLSocketInputRecord.handleUnknownRecord(SSLSocketInputRecord.java:455) ~[?:1.8.0_332]
        at sun.security.ssl.SSLSocketInputRecord.decode(SSLSocketInputRecord.java:184) ~[?:1.8.0_332]
        at sun.security.ssl.SSLTransport.decode(SSLTransport.java:109) ~[?:1.8.0_332]
        at sun.security.ssl.SSLSocketImpl.decode(SSLSocketImpl.java:1397) ~[?:1.8.0_332]
        at sun.security.ssl.SSLSocketImpl.readHandshakeRecord(SSLSocketImpl.java:1305) ~[?:1.8.0_332]
        at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:440) ~[?:1.8.0_332]
        at sun.security.ssl.SSLSocketImpl.ensureNegotiated(SSLSocketImpl.java:818) ~[?:1.8.0_332]
        at sun.security.ssl.SSLSocketImpl.access$200(SSLSocketImpl.java:73) ~[?:1.8.0_332]
        at sun.security.ssl.SSLSocketImpl$AppInputStream.read(SSLSocketImpl.java:909) ~[?:1.8.0_332]
        at java.io.BufferedInputStream.fill(BufferedInputStream.java:246) ~[?:1.8.0_332]
        at java.io.BufferedInputStream.read1(BufferedInputStream.java:286) ~[?:1.8.0_332]
        at java.io.BufferedInputStream.read(BufferedInputStream.java:345) ~[?:1.8.0_332]
        at org.apache.thrift.transport.TIOStreamTransport.read(TIOStreamTransport.java:127) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TTransport.readAll(TTransport.java:86) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslTransport.receiveSaslMessage(TSaslTransport.java:178) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslServerTransport.handleSaslStartMessage(TSaslServerTransport.java:125) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslTransport.open(TSaslTransport.java:271) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslServerTransport.open(TSaslServerTransport.java:41) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslServerTransport$Factory.getTransport(TSaslServerTransport.java:216) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        ... 10 more
2022-08-03T14:48:56,839 ERROR [pool-8-thread-35] server.TThreadPoolServer: Error occurred during processing of message.
java.lang.RuntimeException: org.apache.thrift.transport.TTransportException: javax.net.ssl.SSLException: Unsupported or unrecognized SSL message
        at org.apache.thrift.transport.TSaslServerTransport$Factory.getTransport(TSaslServerTransport.java:219) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.hadoop.hive.metastore.security.HadoopThriftAuthBridge$Server$TUGIAssumingTransportFactory$1.run(HadoopThriftAuthBridge.java:694) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.hadoop.hive.metastore.security.HadoopThriftAuthBridge$Server$TUGIAssumingTransportFactory$1.run(HadoopThriftAuthBridge.java:691) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_332]
        at javax.security.auth.Subject.doAs(Subject.java:360) ~[?:1.8.0_332]
        at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1709) ~[hadoop-common-3.1.1-TDP-0.1.0-SNAPSHOT.jar:?]
        at org.apache.hadoop.hive.metastore.security.HadoopThriftAuthBridge$Server$TUGIAssumingTransportFactory.getTransport(HadoopThriftAuthBridge.java:691) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.server.TThreadPoolServer$WorkerProcess.run(TThreadPoolServer.java:269) [hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_332]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_332]
        at java.lang.Thread.run(Thread.java:750) [?:1.8.0_332]
Caused by: org.apache.thrift.transport.TTransportException: javax.net.ssl.SSLException: Unsupported or unrecognized SSL message
        at org.apache.thrift.transport.TIOStreamTransport.read(TIOStreamTransport.java:129) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TTransport.readAll(TTransport.java:86) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslTransport.receiveSaslMessage(TSaslTransport.java:178) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslServerTransport.handleSaslStartMessage(TSaslServerTransport.java:125) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslTransport.open(TSaslTransport.java:271) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslServerTransport.open(TSaslServerTransport.java:41) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslServerTransport$Factory.getTransport(TSaslServerTransport.java:216) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        ... 10 more
Caused by: javax.net.ssl.SSLException: Unsupported or unrecognized SSL message
        at sun.security.ssl.SSLSocketInputRecord.handleUnknownRecord(SSLSocketInputRecord.java:455) ~[?:1.8.0_332]
        at sun.security.ssl.SSLSocketInputRecord.decode(SSLSocketInputRecord.java:184) ~[?:1.8.0_332]
        at sun.security.ssl.SSLTransport.decode(SSLTransport.java:109) ~[?:1.8.0_332]
        at sun.security.ssl.SSLSocketImpl.decode(SSLSocketImpl.java:1397) ~[?:1.8.0_332]
        at sun.security.ssl.SSLSocketImpl.readHandshakeRecord(SSLSocketImpl.java:1305) ~[?:1.8.0_332]
        at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:440) ~[?:1.8.0_332]
        at sun.security.ssl.SSLSocketImpl.ensureNegotiated(SSLSocketImpl.java:818) ~[?:1.8.0_332]
        at sun.security.ssl.SSLSocketImpl.access$200(SSLSocketImpl.java:73) ~[?:1.8.0_332]
        at sun.security.ssl.SSLSocketImpl$AppInputStream.read(SSLSocketImpl.java:909) ~[?:1.8.0_332]
        at java.io.BufferedInputStream.fill(BufferedInputStream.java:246) ~[?:1.8.0_332]
        at java.io.BufferedInputStream.read1(BufferedInputStream.java:286) ~[?:1.8.0_332]
        at java.io.BufferedInputStream.read(BufferedInputStream.java:345) ~[?:1.8.0_332]
        at org.apache.thrift.transport.TIOStreamTransport.read(TIOStreamTransport.java:127) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TTransport.readAll(TTransport.java:86) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslTransport.receiveSaslMessage(TSaslTransport.java:178) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslServerTransport.handleSaslStartMessage(TSaslServerTransport.java:125) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslTransport.open(TSaslTransport.java:271) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslServerTransport.open(TSaslServerTransport.java:41) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslServerTransport$Factory.getTransport(TSaslServerTransport.java:216) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        ... 10 more
2022-08-03T14:48:57,854 ERROR [pool-8-thread-36] server.TThreadPoolServer: Error occurred during processing of message.
java.lang.RuntimeException: org.apache.thrift.transport.TTransportException: javax.net.ssl.SSLException: Unsupported or unrecognized SSL message
        at org.apache.thrift.transport.TSaslServerTransport$Factory.getTransport(TSaslServerTransport.java:219) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.hadoop.hive.metastore.security.HadoopThriftAuthBridge$Server$TUGIAssumingTransportFactory$1.run(HadoopThriftAuthBridge.java:694) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.hadoop.hive.metastore.security.HadoopThriftAuthBridge$Server$TUGIAssumingTransportFactory$1.run(HadoopThriftAuthBridge.java:691) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_332]
        at javax.security.auth.Subject.doAs(Subject.java:360) ~[?:1.8.0_332]
        at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1709) ~[hadoop-common-3.1.1-TDP-0.1.0-SNAPSHOT.jar:?]
        at org.apache.hadoop.hive.metastore.security.HadoopThriftAuthBridge$Server$TUGIAssumingTransportFactory.getTransport(HadoopThriftAuthBridge.java:691) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.server.TThreadPoolServer$WorkerProcess.run(TThreadPoolServer.java:269) [hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_332]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_332]
        at java.lang.Thread.run(Thread.java:750) [?:1.8.0_332]
Caused by: org.apache.thrift.transport.TTransportException: javax.net.ssl.SSLException: Unsupported or unrecognized SSL message
        at org.apache.thrift.transport.TIOStreamTransport.read(TIOStreamTransport.java:129) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TTransport.readAll(TTransport.java:86) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslTransport.receiveSaslMessage(TSaslTransport.java:178) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslServerTransport.handleSaslStartMessage(TSaslServerTransport.java:125) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslTransport.open(TSaslTransport.java:271) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslServerTransport.open(TSaslServerTransport.java:41) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslServerTransport$Factory.getTransport(TSaslServerTransport.java:216) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        ... 10 more
Caused by: javax.net.ssl.SSLException: Unsupported or unrecognized SSL message
        at sun.security.ssl.SSLSocketInputRecord.handleUnknownRecord(SSLSocketInputRecord.java:455) ~[?:1.8.0_332]
        at sun.security.ssl.SSLSocketInputRecord.decode(SSLSocketInputRecord.java:184) ~[?:1.8.0_332]
        at sun.security.ssl.SSLTransport.decode(SSLTransport.java:109) ~[?:1.8.0_332]
        at sun.security.ssl.SSLSocketImpl.decode(SSLSocketImpl.java:1397) ~[?:1.8.0_332]
        at sun.security.ssl.SSLSocketImpl.readHandshakeRecord(SSLSocketImpl.java:1305) ~[?:1.8.0_332]
        at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:440) ~[?:1.8.0_332]
        at sun.security.ssl.SSLSocketImpl.ensureNegotiated(SSLSocketImpl.java:818) ~[?:1.8.0_332]
        at sun.security.ssl.SSLSocketImpl.access$200(SSLSocketImpl.java:73) ~[?:1.8.0_332]
        at sun.security.ssl.SSLSocketImpl$AppInputStream.read(SSLSocketImpl.java:909) ~[?:1.8.0_332]
        at java.io.BufferedInputStream.fill(BufferedInputStream.java:246) ~[?:1.8.0_332]
        at java.io.BufferedInputStream.read1(BufferedInputStream.java:286) ~[?:1.8.0_332]
        at java.io.BufferedInputStream.read(BufferedInputStream.java:345) ~[?:1.8.0_332]
        at org.apache.thrift.transport.TIOStreamTransport.read(TIOStreamTransport.java:127) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TTransport.readAll(TTransport.java:86) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslTransport.receiveSaslMessage(TSaslTransport.java:178) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslServerTransport.handleSaslStartMessage(TSaslServerTransport.java:125) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslTransport.open(TSaslTransport.java:271) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslServerTransport.open(TSaslServerTransport.java:41) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        at org.apache.thrift.transport.TSaslServerTransport$Factory.getTransport(TSaslServerTransport.java:216) ~[hive-exec-3.1.3-TDP-0.1.0-SNAPSHOT.jar:3.1.3-TDP-0.1.0-SNAPSHOT]
        ... 10 more
```

**Comment by Nuttymoon** [comment]
> @Nuttymoon thank you to open this issue that I did never met before. I confirm having the same error in my env. Do you want me to investigate more, or you are doing it ?

I will not have time to look at it before a few weeks so feel free to investigate ! Thanks @mehdibn

**Comment by mehdibn** [comment]
That's ok, you are welcome :) !

**Comment by mehdibn** [comment]
As discussed, `spark-sql` command part of `spark-thrift-server` is not supported in actual version of TDP.
A new issue will be created in order to remove it.

**Comment by gboutry** [comment]
@mehdibn @Nuttymoon 

After some investigation, I found the root cause of the issue. The problem is that we enable both SSL and SASL on the metaststore. The protocols are exclusive. 

I tested disabling SSL to keep only SASL, and I no longer have connection issues with `spark3-sql`. But then, I quickly encountered impersonation issues which are not related. (Related issue: https://github.com/TOSIT-IO/tdp-collection/issues/347)
